src/server/gc_worker/gc_worker.rs

```diff
@@ -245,11 +235,23 @@ impl<E: Engine> GcRunner<E> {
                 break;
             }
 
+            let mut txn = MvccTxn::for_scan(
+                self.engine.snapshot_on_kv_engine(start_key, end_key)?,
+                Some(ScanMode::Forward),
+                TimeStamp::zero(),
+                false,
+            );
             for key in keys {
-                if let Err(e) = self.gc_key(safe_point, &key) {
+                if let Err(e) = self.gc_key(safe_point, &key, &mut txn) {
                     error!("gc fail"; "key" => %key, "err" => ?e);
                 }
             }
+            let write_size = txn.write_size();
+            let modifies = txn.into_modifies();
+            if !modifies.is_empty() {
+                self.limiter.blocking_consume(write_size);
+                self.engine.modify_on_kv_engine(modifies)?;
+            }
```

Comment:
I think this is not correct.
`is_complete` field returned by `MvccTxn::gc` indicates whether the given key is fully garbage collected. When the deleted keys reached the limit of batch size of `MvccTxn`, (which is `MAX_TXN_WRITE_SIZE = 32 * 1024`), it will break and return early, returning `is_complete = false`. So in `gc_key` function, the loop will never stop once the deleted keys reaches MAX_TXN_WRITE_SIZE.